doc/developer-notes.md

```diff
@@ -530,7 +530,11 @@ its upstream repository.
 Current subtrees include:
 
 - src/leveldb
-  - Upstream at https://github.com/google/leveldb ; Maintained by Google, but open important PRs to Core to avoid delay
+  - Upstream at https://github.com/google/leveldb ; Maintained by Google, but
+    open important PRs to Core to avoid delay.
+  - When updating the leveldb subtree with upstream changes, please sanity check
+    file descriptor usage; see `SetMaxOpenFiles()` in `dbwrapper.cc` for more
+    details.
```

Comment:
Please add mention of the need to check for consensus-incompatible changes.

Reply:
I'm familiar with the accidental fork that was introduced in 0.8 when LevelDB was first added, but what specifically would you like me to add here? That developers should test their change (e.g. on testnet) when updating LevelDB?

Reply:
Testing is not sufficient. The LevelDB changes need to be reviewed for consensus-compatibility. For example, if there was a bug that caused a record with a particular CRC to not be returned later on, and LevelDB fixed this bug, we would need to reject the bugfix from Bitcoin since it would be a hardfork.

Reply:
Updated here, and you can see the GH rendering at https://github.com/eklitzke/bitcoin/blob/ldb_max_open_files/doc/developer-notes.md